### DISCUSSION TITLE
vips jpegsave on a large PDF captures only the back layer

### DISCUSSION THREAD

**Discussion by dfrumin2** [libvips/libvips#3641] [Q&A]
Hi John - one more from me today :-)

I have a series of PDFs that are built in two layers: a baselayer PDF generated from Adobe Indesign and a 600 dpi transparent TIF that is laid on top at full width/height.  I wanat to get a preview of these PDFs so am using vips jpegsave.

99% of the PDFs are working great - it's amazing.  But I have one where the output JPEG only has the baselayer.  The overlay layer (transparent TIF that we generate with copy/images) bonks and doesn't show.  I'm wondering if you have any ideas or if maybe it's related to the other Q&A I put in today - it's the same TIF.  So maybe the TIF has something that throws the code off?

BTW, do you know of a tool that will let me take a PDF and add transparent TIFs in CMYK in an x,y location using 600 dpi?  We built our own version of this and generally it works but curious if you know of some tool/library that might work better.  I know vips doesn't do pdfsave and we had some issue (can't remember) with image magick.  

THANKS!!!!

**Comment by jcupitt** [comment]
Hi again, yes, I'd guess it was something funny about that TIFF. You'd need to share it somehow, maybe privately if it's customer data.

I don't do much with PDFs, so I don't know of a good library that'd do annotation, sorry.
